### PR TITLE
chore: document next.js specific component template

### PR DESCRIPTION
**Pull request by lmiller1990**
- Closes https://cypress-io.atlassian.net/browse/UNIFY-1608

### User facing changelog

Document Next.js specific selector injected into `component-index.html`.

### Additional details

Prior context: https://github.com/cypress-io/cypress/pull/16247 

Basically, Next.js hard-codes a specific DOM node and expects it to exist for injecting CSS via style-loader. You can see the line of code here: https://github.com/vercel/next.js/blob/5f3351dbb8de71bcdbc91d869c04bc862a25da5f/packages/next/build/webpack/config/blocks/css/loaders/client.ts#L24

We need this too, so Next.js components render the same as the user expects when using Cypress Component Testing.

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/scaffold-config/package.json
Comment (lmiller1990) [inline packages/scaffold-config/package.json:12]: These were not running on CI because this was missing so I added it.

File: system-tests/project-fixtures/next/cypress/support/component-index.html
Comment (lmiller1990) [inline system-tests/project-fixtures/next/cypress/support/component-index.html:5]: Above file packages/scaffold-config/test/unit/frameworks.spec.ts was obsolete (code was moved to `data-context` and tests added there) so I deleted it. The reason we did not notice was because `scaffold-config` was not running on CI (now it is as of this PR).

**Review by marktnoonan — CHANGES_REQUESTED** [review]
This looks good and will help users avoid breaking the index html. It's especially good to get those unit tests running in CI!

One change request - one of the new changes is causing E2E failures in scaffolding component tests, I think due to including the `projectRoot` in the glob? But not too familiar with this.

Unrelated: I noticed when I got in that spec is that there's a TODO [here](https://github.com/cypress-io/cypress/blob/1289b01947a28a77048114aa63cbcd62d9ff52ee/packages/launchpad/cypress/e2e/scaffold-component-testing.cy.ts#L11-L12) where the PR that was referred to has been merged, should we go on and make a ticket for that TODO?

File: packages/scaffold-config/src/frameworks.ts
Comment (ZachJW34) [inline packages/scaffold-config/src/frameworks.ts:141]: It would make more sense to have this logic in `componentIndexHtmlGenerator` since it is aware of the proper spacing required to produce a formatted html, but this is the only use case so not a big deal.

**Review by ZachJW34 — APPROVED** [review]
Code changes make sense, going to approve so once they are fixed up you can merge

File: packages/scaffold-config/src/frameworks.ts
Comment (lmiller1990) [reply packages/scaffold-config/src/frameworks.ts:141]: Not sure I agree here - I think `componentIndexHtmlGenerator` should not be too smart, and we should be injecting the additional markup from the outside. I don't think it needs to know about the different frameworks we support.

I tried to do something like:

```js
componentIndexHtml: dedent`
  <!-- stuff -->
`
```

But it won't work, since you end up with a `dedent` string inside a `dedent` string, and it doesn't format as you'd expect. This was the only way I could use the injection API that was implemented by the original developer and maintain the spacing. Not ideal, but it is what it is 🤷

File: packages/scaffold-config/src/detect.ts
Comment (lmiller1990) [reply packages/scaffold-config/src/detect.ts:138]: Hm good catch - I needed `cwd` for the tests (that were not running) to pass. I'll try your suggestion of removing `projectRoot` from the globs - I think that makes more sense anyway, we should be able to assume that cypress is running in the projectRoot.

Comment (lmiller1990) [reply packages/scaffold-config/src/detect.ts]: Fixed.

**Comment by lmiller1990** [comment]
> This looks good and will help users avoid breaking the index html. It's especially good to get those unit tests running in CI!
> 
> One change request - one of the new changes is causing E2E failures in scaffolding component tests, I think due to including the `projectRoot` in the glob? But not too familiar with this.
> 
> Unrelated: I noticed when I got in that spec is that there's a TODO [here](https://github.com/cypress-io/cypress/blob/1289b01947a28a77048114aa63cbcd62d9ff52ee/packages/launchpad/cypress/e2e/scaffold-component-testing.cy.ts#L11-L12) where the PR that was referred to has been merged, should we go on and make a ticket for that TODO?

I fixed the tests 👍 

Not sure about the TODO, we could, or it might be already captured in some other tests. Likely a low priority.

File: packages/scaffold-config/src/detect.ts
Comment (lmiller1990) [reply packages/scaffold-config/src/detect.ts:138]: I think we can just use the `async` call of `globby` to avoid EMFILE errors. Unfortunately, making this function as `async` has a huge cascading effect requiring 10s of files to be changed. We end up calling a function (that now needs to be `awaited` in a few constructors, too, which makes it more painful, since a constructor cannot be `async`, the design and flow of some parts of the system would need to be reconsidered.

We should still make this change, it's just not as trivial as "use await, mark async".

**Pull request merged**
